docs/NNNN-schema-registry.md

```diff
@@ -0,0 +1,152 @@
+# OCFL Community Extension - Schema Registry
+
+Extension Name: NNNN-schema-registry
+Authors: P. Cornwell, D. Granville
+Minimum OCFL Version: 1.0
+OCFL Community Extensions Version: 1.0
+Extension Version: 0.1
+Obsoletes: n/a
+Obsoleted by: n/a
+
+## Overview
+An OCFL object will typically contain metadata serialised in JSON or XML files. These will normally conform to one or more specified schemata and reference the external JSONSchema/XSD/DTD.
+
+In order for an OCFL root to represent a self-consistent repository, the specific versions of each schema referenced by content must be available. Maintaining a local copy of the schema is therefore prudent, especially where longer-term preservation is a goal and external online references cannot be relied upon.
+
+This extension stores a copy of each schema referenced by the content of OCFL objects, thus avoiding the need to store a copy of the schema redudantly in each OCFL object. It also provides a convenient reference point for archive management software and future users inspecting the contents of the OCFL object.
+
+We describe a standardised layout for a schema repository - essentially implemented as a 'meta' OCFL object: a sub-directory containing local copies of schemata. This directory contains all schemata, with standard OCFL integrity checks, in use within the OCFL root.
+
+## Parameters
+
+Configuration is done by setting appropriate values in config.json at the top level of the extension's directory. The keys expected are:
+
+extensionName
+: String identifying the extension. Must be "NNNN-schema-registry".
+
+schemaDigestAlgorithm
+: Algorithm used for calculating safe filenames from schema identifiers. "md5" is the default.
+
+Additionally a schema_registry.json file should be present. This contains the keys -
+
+digestAlgorithm
```

Comment:
What is the advantage to using multiple digest algorithms?

Reply:
Since the schema identifiers are merely string-sized, a smaller digest seemed reasonable. I don't think this makes implementing the extension any more difficult? 

However, this did make me think - a weaker digest like MD5 coud potentially be expolited with maliciously crafted identifiers that colide with existing ones. It seems highly unlikley, but I've added a note about checking for this. 

If others feel that using the same algorithm for file fixity and the local filenames is preferable, we're happy to do it that way.